### PR TITLE
Test installer commands

### PR DESCRIPTION
**Pull request by jonathanhefner**
This PR _looks_ large because this project currently has no testing infrastructure.  The vast majority of LOC in this PR are from `Gemfile` and `Gemfile.lock` files.  I have split the PR into multiple commits to further clarify.

---

This adds test coverage for the `javascript:install:bun`, `javascript:install:esbuild`, `javascript:install:rollup`, and `javascript:install:webpack` commands.  The commands are tested against a freshly generated Rails app using the version of Rails that is currently loaded.  Thus the installer can be tested with different versions of Rails in CI.

Infrastructure has been added to run tests against Rails 7.0, 7.1, and `main` branch.

This PR takes a different approach than https://github.com/rails/dartsass-rails/pull/49 — instead of separate appraisals / gemfiles for Sprockets and Propshaft, this PR has tests that target each.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+          - "3.1"
+          - "3.2"
+          - "3.3"
+        gemfile:
+          - gemfiles/rails_7_0_propshaft.gemfile
+          - gemfiles/rails_7_1_propshaft.gemfile
+          - gemfiles/rails_main_propshaft.gemfile
+          - gemfiles/rails_7_0_sprockets.gemfile
+          - gemfiles/rails_7_1_sprockets.gemfile
+          - gemfiles/rails_main_sprockets.gemfile
```
Comment (jonathanhefner) [inline .github/workflows/ci.yml:18]: This only tests with Rails >= 7.0 and Ruby >= 3.1.  I can add entries for Rails 6.0 and 6.1, and older versions of Ruby, but first I wanted to confirm which versions we want to support.

File: Gemfile [hunk 2]
```diff
@@ -4,9 +4,8 @@ git_source(:github) { |repo| "https://github.com/#{repo}.git" }
 # Specify your gem's dependencies in importmap-rails.gemspec.
 gemspec
 
-rails_version = ENV["RAILS_VERSION"] || "6.1.0"
-gem "rails", "~> #{rails_version}"
-
+gem "appraisal"
+gem "rails", "~> 6.0.0"
```
Comment (jonathanhefner) [inline Gemfile:8]: This file defaulted the Rails version to 6.1, but the [gemspec specifies 6.0](https://github.com/rails/jsbundling-rails/blob/44919fe2ecc11fd23b26fc39692615d87656da7f/jsbundling-rails.gemspec#L14).  Which is correct?  (Or should we jump to Rails 7.0?)

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+      fail-fast: false
+      matrix:
+        ruby-version:
+          - "3.1"
+          - "3.2"
+          - "3.3"
+        gemfile:
+          - gemfiles/rails_7_0.gemfile
+          - gemfiles/rails_7_1.gemfile
+          - gemfiles/rails_main.gemfile
```
Comment (jonathanhefner) [inline .github/workflows/ci.yml:15]: This only tests with Rails >= 7.0 and Ruby >= 3.1. I can add entries for Rails 6.0 and 6.1, and older versions of Ruby, but first I wanted to confirm which versions we want to support.

File: Gemfile [hunk 4]
```diff
@@ -4,9 +4,9 @@ git_source(:github) { |repo| "https://github.com/#{repo}.git" }
...
```
Comment (jonathanhefner) [inline Gemfile:8]: This file defaulted the Rails version to 6.1, but the [gemspec specifies 6.0](https://github.com/rails/jsbundling-rails/blob/44919fe2ecc11fd23b26fc39692615d87656da7f/jsbundling-rails.gemspec#L14).  Which is correct?  (Or do we want to jump to Rails 7.0?)

File: .github/workflows/ci.yml (hunk 3 above)
Comment (jonathanhefner) [inline .github/workflows/ci.yml:15]: This only tests with Rails >= 7.0 and Ruby >= 3.1. I can add entries for Rails 6.0 and 6.1, and older versions of Ruby, but first I wanted to confirm which versions we want to support.

File: Gemfile (hunk 4 above)
Comment (dhh) [reply Gemfile:8]: I believe we should retain compatibility with 6.0.

File: .github/workflows/ci.yml (hunk 3 above)
Comment (dhh) [reply .github/workflows/ci.yml:15]: I think it's fine to focus on these for now.

**Pull request merged**